refinery/core/views.py

```diff
@@ -1040,6 +1041,68 @@ def post(self, request):
         return Response(token)
 
 
+class SiteProfileViewSet(APIView):
+    """API endpoint that allows for UserProfiles to be edited.
+     ---
+    #YAML
+
+    PATCH:
+        parameters_strategy:
+        form: replace
+        query: merge
+
+        parameters:
+            - name: about_markdown
+              description: Markdown paragraph
+              type: string
+              paramType: form
+              required: false
+            - name: intro_markdown
+              description: Markdown paragraph
+              type: string
+              paramType: form
+              required: false
+            - name: twitter_username
+              description: twitter user name to display twitter feed
+              type: string
+              paramType: form
+              required: false
+            - name: yt_videos
+              description: string object with video ids and captions
+              type: string
+              paramType: form
+              required: false
+    ...
+    """
+    http_method_names = ["get", "patch"]
+
+    def get(self, request):
+        site_profile = SiteProfile.objects.get(site=get_current_site(request))
+        serializer = SiteProfileSerializer(site_profile)
+        return Response(serializer.data)
+
+    def patch(self, request):
+        if not request.user.is_superuser():
```

Comment:
I'm sure you're thinking of adding this, but a test to ensure admin-only access would be great